File: Cpp/1816_Truncate_Sentence.cpp
class Solution {
public:
    string truncateSentence(string s, int k) {
        string result = "";
        int i = 0;
        for(auto& ch : s){
            if(ch == ' ') ++i;
            if(k == i) return result;
            result += ch;
        }
        
        return result;
    }
};

File: Kotlin/1816_Truncate_Sentence.kt
class Solution {
    fun truncateSentence(s: String, k: Int): String {
        val result = StringBuilder()
        var i = 0
        
        for(ch in s){
            if(ch == ' ') ++i
            if(k == i) return result.toString()
            result.append(ch)
        }
        
        return result.toString()
    }
}